src/services/codefixes/fixAddMissingMember.ts

```diff
@@ -405,4 +429,87 @@ namespace ts.codefix {
+        const props = map(info.properties, prop => {
+            const initializer = prop.valueDeclaration
+                ? tryGetInitializerValueFromType(context, checker, importAdder, quotePreference, checker.getTypeAtLocation(prop.valueDeclaration))
+                : createUndefined();
+            return factory.createPropertyAssignment(prop.name, initializer);
+        });
+        changes.replaceNode(context.sourceFile, info.parentDeclaration, factory.createObjectLiteralExpression([...info.parentDeclaration.properties, ...props], /*multiLine*/ true));
+    }
+
+    function tryGetInitializerValueFromType(context: CodeFixContextBase, checker: TypeChecker, importAdder: ImportAdder, quotePreference: QuotePreference, type: Type): Expression {
+        if (type.flags & TypeFlags.AnyOrUnknown) {
+            return createUndefined();
+        }
+        if (type.flags & TypeFlags.String) {
+            return factory.createStringLiteral("", /* isSingleQuote */ quotePreference === QuotePreference.Single);
+        }
+        if (type.flags & TypeFlags.Number) {
+            return factory.createNumericLiteral(0);
+        }
+        if (type.flags & TypeFlags.BigInt) {
+            return factory.createBigIntLiteral("0n");
+        }
+        if (type.flags & TypeFlags.Boolean) {
+            return factory.createFalse();
+        }
+        if (type.flags & TypeFlags.EnumLike) {
+            const enumMember = type.symbol.exports ? firstOrUndefined(arrayFrom(type.symbol.exports.values())) : type.symbol;
+            const name = checker.symbolToExpression(type.symbol.parent ? type.symbol.parent : type.symbol, SymbolFlags.Value, /*enclosingDeclaration*/ undefined, /*flags*/ undefined);
+            return enumMember === undefined || name === undefined ? factory.createNumericLiteral(0) : factory.createPropertyAccessExpression(name, checker.symbolToString(enumMember));
+        }
+        if (type.flags & TypeFlags.NumberLiteral) {
+            return factory.createNumericLiteral((type as NumberLiteralType).value);
+        }
+        if (type.flags & TypeFlags.BigIntLiteral) {
+            return factory.createBigIntLiteral((type as BigIntLiteralType).value);
+        }
+        if (type.flags & TypeFlags.StringLiteral) {
+            return factory.createStringLiteral((type as StringLiteralType).value, /* isSingleQuote */ quotePreference === QuotePreference.Single);
+        }
+        if (type.flags & TypeFlags.BooleanLiteral) {
+            return (type === checker.getFalseType() || type === checker.getFalseType(/*fresh*/ true)) ? factory.createFalse() : factory.createTrue();
+        }
+        if (type.flags & TypeFlags.Null) {
+            return factory.createNull();
+        }
+        if (type.flags & TypeFlags.Union) {
+            const expression = firstDefined((type as UnionType).types, t => tryGetInitializerValueFromType(context, checker, importAdder, quotePreference, t));
+            return expression ?? createUndefined();

```

Comment:
Sorry, what I meant is that not only will you potentially get `undefined` (which might not be the case anymore anyway), but you might also get the result of `createUndefined()` even if another result might be more appropriate. For example, with `undefined | number`.

However, thinking about it some more, maybe that's okay.

Reply:
For the similar case `j: undefined | "special-string"` in `codeFixAddMissingProperties1` we will get `"special-string"`

Reply:
Yeah, that's the case I had in mind. Both are fine, we should just be intentional in the behavior and ensure it's tested.